addons/stock/migrations/11.0.1.1/post-migration.py

```diff
@@ -189,11 +176,164 @@ def create_stock_move_line(env):
                 ON smol.operation_id = spo.id
             INNER JOIN stock_move sm ON sm.id = smol.move_id
             INNER JOIN product_product pp ON spo.product_id = pp.id
+            LEFT JOIN product_template pt ON pt.id = pp.product_tmpl_id
             LEFT JOIN stock_picking sp ON sp.id = spo.picking_id
-            LEFT JOIN stock_quant sq ON sq.id = smol.reserved_quant_id
-            LEFT JOIN stock_production_lot spl ON spl.id = sq.lot_id
-        WHERE sm.state = 'done'
-        """, (AsIs(lot_expr), ),
+        WHERE
+            sm.state = 'done' AND
+            sm.scrapped IS NOT true AND
+            pt.tracking NOT IN ('serial', 'lot') AND
+            smol.qty > 0
+        """,
+    )
+
+
+@openupgrade.logging()
+def create_stock_move_line_done_tracked(env):
+    """This method creates stock.move.line recreated from old done
+    stock.pack.operation.lot records for products with tracking by lot/serial.
+
+    Lot information cannot be taken from stock.move.operation.link using the field
+    'reserved_quand_id' because it is a dead field.
+    stock.pack.operation/stock.operation.lot as v10 grouped together several
+    stock.move lines with the same product in the same operation record, making
+    it very difficult/time consuming to split the stock.pack.operation.lot across
+    each move.
+    So this method has a known limitation: if, on a picking, there are several
+    stock.move with the same product (not so common...), the
+    stock.move.line created by the migration will all be attached to the same
+    stock.move. Therefore, in this particular case, the sum of the qty of the
+    stock.move.line will be superior to the qty of the stock.move.
```

Comment:
Thanks for mentioning this. I don't know if having more than one stock move on the same picking with the same product is that rare. In fact, it was the goal of Pedro's earlier PR to fix this: https://github.com/OCA/OpenUpgrade/commit/91bba0cf0f4aac58ff812f59f38d7ea28eda7634

Would it be possible to check for these cases after this query and correct them using ORM operations? It woul then only be a performance hit for those databases for which the operation is necessary.